timesketch/api/v1/resources/sigma.py

```diff
@@ -90,3 +93,39 @@ def get(self, rule_uuid):
                 HTTP_STATUS_CODE_NOT_FOUND, 'No sigma rule found with this ID.')
 
         return return_rule
+
+class SigmaByTextResource(resources.ResourceMixin, Resource):
+    """Resource to get list of Sigma rules."""
+
+    @login_required
+    def post(self):
+        """Handles POST request to the resource.
+
+        Returns:
+            JSON sigma rule
+        """
+        form = forms.SigmaRuleForm.build(request)
+        # optional: do we want to have a title at this point?
+        title = form.title.data
+        content = form.content.data
+
+        print(f'Content passed: {content}')
+        if not form.validate_on_submit():
+            abort(HTTP_STATUS_CODE_BAD_REQUEST, 'Unable to validate form data')
+
+        try:
+            sigma_rule = ts_sigma_lib.get_sigma_rule_by_text(content)
+
+        except ValueError:
+            # TODO: make better responses here with different scenarios.
+            logger.error('Parsing error',
+                         exc_info=True)
+            abort(
+                HTTP_STATUS_CODE_NOT_FOUND,
+                'Error unable to parse the provided Sigma rule')
+
+        if sigma_rule is None:
+            abort(
+                HTTP_STATUS_CODE_NOT_FOUND, 'No sigma was parsed')
```

Comment:
is that the case? or that the sigma rule was somehow not properly formed? or would that always raise an exception?

Reply:
now with results other than ```None```  a much more granular response can be given to the user.

Reply:
and this is basically last resort, It should never be reached, but I will keep it in the code